responsibleai/responsibleai/feature_metadata.py

```diff
@@ -1,57 +1,66 @@
         """Placeholder class for feature metadata provided by the user.
 
         :param identity_feature_name: Name of the feature which helps to
                                       uniquely identify a row or instance
                                       in user input dataset.
         :type identity_feature_name: Optional[str]
-        :param datetime_features: List of datetime features in the
-                                  user input dataset.
+        :param datetime_features: names of datetime features in the user input
+                                  dataset.
         :type datetime_features: Optional[List[str]]
         :param categorical_features: List of categorical features in the
                                      user input dataset.
         :type categorical_features: Optional[List[str]]
         :param dropped_features: List of features that were dropped by the
                                  the user during training of their model.
         :type dropped_features: Optional[List[str]]
+        :param time_series_id_features: List of features that are used
+                                        to uniquely identify a time
+                                        series in the user input dataset.
+        :type time_series_id_features: Optional[List[str]]
         """
         self.identity_feature_name = identity_feature_name
         self.datetime_features = datetime_features
         self.categorical_features = categorical_features
         self.dropped_features = dropped_features
-        if self.datetime_features is not None:
-            warnings.warn('datetime_features are not in use currently.')
-        if self.categorical_features is not None:
-            warnings.warn('categorical_features are not in use currently.')
+        self.time_series_id_features = time_series_id_features
 
-    def validate_feature_metadata_with_user_features(
-            self, user_features: Optional[List[str]] = None):
-        """Validate the feature metadata with the user features.
+    def validate(self, feature_names: List[str]):
+        """Validate the user-provided feature metadata.
 
-        :param user_features: List of features in the user input dataset.
-        :type user_features: Optional[List[str]]
+        :param feature_names: list of features in the dataset.
+        :type feature_names: List[str]
         """
-        if user_features is None:
-            return
-        if self.identity_feature_name is not None:
-            if self.identity_feature_name not in user_features:
-                raise UserConfigValidationException(
-                    'The given identity feature name {0} is not present'
-                    ' in user features.'.format(
-                        self.identity_feature_name))
+        identity_feature = ([self.identity_feature_name]
+                            if self.identity_feature_name else None)
+        datetime_features = (self.datetime_features
```

Comment:
I believe this is a remnant from having it as a single time column. In that case I needed to convert it to a list to make it compatible with the validation function. So that means you're absolutely right and I will adjust the code 🙂  Nice catch!